core/src/main/scala/org/apache/spark/h2o/utils/ReflectionUtils.scala

```diff
@@ -173,7 +176,7 @@ case class ProductMember(name: String, typeName: NameOfType) {
   override def toString = s"$name: $typeName"
 }
 
-case class ProductType(members: Array[ProductMember]) {
+case class ProductType(members: Array[ProductMember], types: Array[Class[_]]) {
```

Comment:
Why an additional argument? Let's make these "types" (actually, classes) members of ProductMember. That's what that class is for.